sdk/src/types/block/address/mod.rs

```diff
@@ -168,6 +168,9 @@ impl Address {
             }
             // TODO maybe shouldn't be a semantic error but this function currently returns a TransactionFailureReason.
             (Self::Anchor(_), _) => return Err(TransactionFailureReason::SemanticValidationFailed),
+            (Self::ImplicitAccountCreation(ImplicitAccountCreationAddress(ed25519_address)), _) => {
+                Address::from(ed25519_address.clone()).unlock(unlock, context)?;
```

Comment:
Doesn't need to be added to inserted in `context.unlocked_addresses`?

Reply:
It will, but in the `Self::Ed25519` match of the inner address. Otherwise we're adding the ImplicitAccountCreationAddress as unlocked, which could cause issues

Reply:
`but in the Self::Ed25519 match of the inner address.`
Can you point me to the code? Not sure what you mean here

Reply:
maybe it's worth a comment?

Reply:
Ah it's calling the same function, I thought the unlock was on another type, all good then